### ISSUE TITLE
redirect_uri refusing to be recognized as matching Authorization callback URL at github

### ISSUE THREAD

**Issue by robbiemu** [lelylan/simple-oauth2#71]
I'm implementing the github example in a project template generated in intellij idea. I am getting this in the log:
```
GET /auth/github 302 3.791 ms - 458
Simple OAuth2 : OAuth2 Node Request
Simple OAuth2 : Making the HTTP request { uri: 'https://github.com/login/oauth/access_token',
  method: 'POST',
  headers: { Authorization: 'Basic SOME_LONG_RANDOM_BASE64_LOOKING_STRING==' },
  form: 
   { code: undefined,
     redirect_uri: 'http://ANON_IP:3000/auth/callback/github',
     grant_type: 'authorization_code',
     client_id: //MYCLIENT_ID//.
     client_secret: //MYCLIENT_SECRET// } }
Simple OAuth2 : Checking response body error=bad_verification_code&error_description=The+code+passed+is+incorrect+or+expired.&error_uri=https%3A%2F%2Fdeveloper.github.com%2Fv3%2Foauth%2F%23bad-verification-code
```
_(I've anonymized the id and secret, and the redirect_uri ip of course)_
in a short while a new GET response comes in the server logs:
```
GET /auth/callback/github?error=redirect_uri_mismatch&error_description=The+redirect_uri+MUST+match+the+registered+callback+URL+for+this+application.&error_uri=https%3A%2F%2Fdeveloper.github.com%2Fv3%2Foauth%2F%23redirect-uri-mismatch&state=3%28%230%2F%21%7E - - ms - -
```

my callback url in the code is reflected in the redirect_uri field in the log above. I anonymized my ip address there but trust me  have verified it (and receive the error callback on it). my "Authorization Callback URL" in my github registered app is identical.

PS> please consider changing the example code so that it handles the ?error in the callback. This changes the behavior of the app, at first I thought I was having two problems, the main one being something setting a code field before requesting it from the server. properly outputting an error like with the following makes it much clearer what is wrong:
```
    if(req.query.hasOwnProperty("error")){
        var err = new Error(req.query.error);
        err.status = req.query.error_description.replace(/\+/," ");
        err.stack = req.url;
        next(err);
    }
```

**Comment by robbiemu** [comment]
I am able to get around the problem! I followed the advice in the github documentation and commented out the redirect_uri specified in the authentication_uri, and it works. However, it is identical to the "Authorization callback URL" setting in my github setup for the app, so I'm not sure _why_ I had to do that.

How do I clear the token and log a user out?

**Comment by jonathansamines** [comment]
Solved in 1.0.0

**Comment by pranavkumar24** [comment]
robbiemu, as you have mentioned "I am able to get around the problem! I followed the advice in the github documentation and commented out the redirect_uri specified in the authentication_uri, and it works. However, it is identical to the "Authorization callback URL" setting in my github setup for the app, so I'm not sure why I had to do that."
can you please help to understand what and where you have made changes?
